src/core/OIDNDenoiser.js

```diff
@@ -0,0 +1,283 @@
+// Utils to interface with the denoiser
+
+import { Denoiser } from 'denoiser';
+import { FullScreenQuad } from 'three/examples/jsm/postprocessing/Pass.js';
+import { ClampedInterpolationMaterial } from '../materials/fullscreen/ClampedInterpolationMaterial.js';
+
+import { NoBlending, WebGLRenderTarget, MeshBasicMaterial, LinearSRGBColorSpace } from 'three';
+
+export class OIDNDenoiser {
+
+	get quality() {
+
+		return this.denoiser.quality;
+
+	}
+
+	set quality( v ) {
+
+		this.denoiser.quality = v;
+
+	}
+
+	get useAux() {
+
+		return this._useAux;
+
+	}
+
+	set useAux( v ) {
+
+		this._useAux = v;
+		this.denoiser.resetInputs();
+
+	}
+
+	get cleanAux() {
+
+		return ! this.denoiser.dirtyAux;
+
+	}
+
+	set cleanAux( v ) {
+
+		this.denoiser.dirtyAux = ! v;
+
+	}
+
+	get hdr() {
+
+		return this.denoiser.hdr;
+
+	}
+
+	set hdr( v ) {
+
+		this.denoiser.hdr = v;
+
+	}
+
+	get weightsUrl() {
+
+		return this.denoiser.weightsUrl;
+
+	}
+
+	set weightsUrl( url ) {
+
+		this.denoiser.weightsUrl = url;
+
+	}
+
+	// Debugging
+	get denoiserDebugging() {
+
+		return this.denoiser.debugging;
+
+	}
+
+	set denoiserDebugging( v ) {
+
+		this.denoiser.debugging = v;
+
+	}
+
+	constructor( renderer ) {
+
+		this.renderer = renderer;
+		this.denoiser = new Denoiser( 'webgl', renderer.domElement );
+		this.denoiser.inputMode = 'webgl';
+		this.denoiser.outputMode = 'webgl';
+		this.denoiser.weightsUrl = 'https://cdn.jsdelivr.net/npm/denoiser/tzas';
+		this.denoiser.onProgress( progress => this.handleProgress( progress ) );
+		//this.denoiser.hdr = true;
+		this.denoiser.webglStateManager.ingoreRestore = true;
```

Comment:
spelling issue - ingoreRestore -> ignoreRestore. Can we add an inline comment describing why we need this